### ISSUE TITLE
Release/ffi_bindings.node: undefined symbol: napi_set_instance_data\n"

### ISSUE THREAD

**Issue by Supriyaj29** [ffi-cross/js-ffi-cross#7]
After failed attempt of using  ffi-napi, I decided to use ffi-cross, hoping that it will fix the below error. 

symbol lookup error: node_modules/ffi-cross/build/Release/ffi_bindings.node: undefined symbol: napi_set_instance_data

I am using node v12.22.6 on Ubuntu LTS 20.04.2. 

My implementation look like this:

`const { ref, ffi } = require('ffi-cross');
const struct = require('ref-struct-napi');
const refarray = require('ref-array-napi');

let retStatus = struct({
    'status': ref.types.int,
    'msg': refarray('char', 128)
});
let statusPtr = ref.refType(retStatus);

let wrapper = ffi.Library('libcustom_request', {
    'send_request': ['int', [statusPtr, 'uint8*']],
});`

Is there anything missing in this implementation? 

I also tried finding out definition for napi_set_instance_data and napi_get_instance_data functions. I checked node_api.h files for both node v12 and node v14. Looked into the documentation as well, but no luck. Would really appreciate if you could share some information on this, or if you any suggestions.

**Comment by Supriyaj29** [comment]
Thank you for quick response.  As suggested I removed said packages. I also following this example given on tutorials page. 
https://github.com/ffi-cross/js-ffi-cross/blob/master/docs/tutorial.md#structs  and accordingly updated my implementation. Still no luck. I am still getting same error.

const { ffi } = require('ffi-cross');
const { ref, types, StructType } = ffi ;

let retStatus = StructType({
'status': types.int,
'msg': types.CString
});
let statusPtr = ref.refType(retStatus);

let wrapper = ffi.Library('libcustom_request', {
'send_request': ['int', [statusPtr, 'uint8*']],
});`

Looking at the error, it kind of make sense, because I looked into nodejs include folder and also the .node_gyp/<version>/node/node_api.h, none of the files has napi_set_instance_data definition.

**Comment by Supriyaj29** [comment]
> Could you share a repository that reproduces the problem? It looks to me like that function is supposed to be defined by node for `NAPI_VERSION >= 6`. I suspect there is a misconfiguration in your build somehow.

Amm.. As I mentioned before, I have added this node_module as a dependency for my application and the implementation is as given in earlier comments. And I am using
node 12.22.6
node-gyp v8.2.0

I also ran node-gyp clean and node-gyp configure --napi_build_version=6 from inside this node_module/ffi-cross/ folder. Even then it failed. So I am thinking of going back to node v12.17 and try again.
